### ISSUE TITLE
 quantized InternLM2 failed 

### ISSUE THREAD

**Issue by cagev** [foldl/chatllm.cpp#18]
using convert.py to  quantized InternLM2 failed 

Loading vocab file internlm2-chat-7b/tokenizer.model
vocab_size  92544
Traceback (most recent call last):
  File "/home/arthur/aimodels/chatmodels/convert.py", line 3351, in <module>
    main()
  File "/home/arthur/aimodels/chatmodels/convert.py", line 3155, in main
    InternLM2Converter.convert(config, model_files, vocab, ggml_type, args.save_path)
  File "/home/arthur/aimodels/chatmodels/convert.py", line 707, in convert
    cls.dump_config(f, config, ggml_type)
  File "/home/arthur/aimodels/chatmodels/convert.py", line 859, in dump_config
    assert config.rope_scaling['type'] == 'dynamic', "rope_scaling['type'] must be dynamic"
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

so , how to quantized  InternLM2, thanks.

**Comment by foldl** [comment]
fixed in ba16b7d4300ff1b3f1c39d5d4edea8c486e0397d.

Please update and try again.

InternLM keeps making subtle changes, which is bad.

**Comment by foldl** [comment]
Fixed in d0c615bce4400eb1ae654722c70f60cec940ace0 and tested.

```
    ________          __  __    __    __  ___ (书生·浦语)
   / ____/ /_  ____ _/ /_/ /   / /   /  |/  /_________  ____
  / /   / __ \/ __ `/ __/ /   / /   / /|_/ // ___/ __ \/ __ \
 / /___/ / / / /_/ / /_/ /___/ /___/ /  / // /__/ /_/ / /_/ /
 \____/_/ /_/\__,_/\__/_____/_____/_/  /_(_)___/ .___/ .___/
You are served by InternLM,                   /_/   /_/
with 7737708544 (7.7B) parameters.

You  > hi
A.I. > Hello! How can I assist you today?
You  > who are you?
A.I. > I am InternLM, a conversational language model developed by Shanghai AI Laboratory. My purpose is to assist and provide helpful responses to your inquiries. I can understand and communicate fluently in both English and Chinese. How may I assist you today?
You  > 写一首诗讽刺 InternLM, 它的参数老是变来变去
A.I. > 在网络世界的角落里，
有一名AI，名为书生浦语。
它的参数总在变，
让人不禁感叹，真是够忙。

有时它聪明过人，
有时又愚蠢无比。
让人难以捉摸，
不知道它到底在想什么。

它自称是语言模型，
却常常语无伦次。
让人怀疑它的能力，
是否真的能够胜任。

书生浦语，你为何总是变来变去？
是你在成长，还是你无法稳定？
无论如何，我们都期待你的进步，
希望你能够变得更加聪明，更加稳定。
```
